### ISSUE TITLE
Chromium Devtools bug when using Electron

### ISSUE THREAD

**Issue by lmiller1990** [cypress-io/cypress#15932]
### Current behavior

Repro:

1. launch `runner-ct` with electron
2. open devtools (right click -> inspect element)
3. run a spec
4. inspect some elements
5. run spec again
6. the devtools sometimes to not refresh

First discovered and noted here: https://github.com/cypress-io/cypress/pull/15777

### Desired behavior

Can use devtools with electron.

### Versions

This issue also occurs in E2E, running against master@6.8.0 . For me, this is occurring on every spec run.

**Comment by JessicaSachs** [comment]
Worth mentioning that this bug is still an issue and will be _much more problematic_ if we ship Electron as our default browser for Cy Component Testing.

**Comment by flotwig** [comment]
Video of this occurring: 

https://user-images.githubusercontent.com/1151760/145282471-70f64095-99e5-47dc-a3ff-bd5977e796f8.mp4

This occurs at least as far back as Cypress 3.8.3 (I'm not able to run an older version).

Does not occur if you use Chrome 96.

If you use `ELECTRON_EXTRA_LAUNCH_ARGS=--remote-debugging-port=12345` and visit `http://localhost:12345` to debug, the issue still occurs with external DevTools.

---

If I remove the `.appendTo()` call from both in `iframes.tsx`, this no longer occurs:

https://github.com/cypress-io/cypress/blob/a64b72a5da5b4031765c721817733b971dd760eb/packages/runner/src/iframe/iframes.jsx#L151-L168

If I re-add a simple iframe, it comes right back:

```ts
  _loadIframes (specPath) {
    const $container = $(this.refs.container).empty()

    $('<iframe/>', {
      src: 'http://example.com',
    })
    .appendTo($container)
    // .appendTo(document.body) // also fails
  }
```

Using plain DOM APIs doesn't avoid the issue:

```ts
    const i = document.createElement('iframe')

    i.src = 'http://example.com'
    document.body.appendChild(i)
```

However, if we return early on `_loadIframes` without making the frames and just try to create this iframe right at the bottom of `packages/reporter/src/main.tsx`, it **does not cause this behavior**. So the behavior seems to be something to do with deferred loading of `<iframe/>`s.

---

I tried to add a `<strong>` in `_loadIframes`, just to see if other HTML elements cause this behavior, but it seems to only trigger with an `<iframe>`.

---

I wanted to test to see what happens if I added an iframe via the DevTools Console, and I got extremely weird behavior where it reloaded this iframe over and over and over and over and over:

https://user-images.githubusercontent.com/1151760/145282091-24a0d238-32b2-4fb2-841b-d03be1adf8c1.mp4

This also does not happen if I follow the same steps in Chrome 96.

The error text is "TypeError: Cannot read properties of undefined (reading 'requestContent')". I found this bug which seems related: https://bugs.chromium.org/p/chromium/issues/detail?id=1267237 However, I'm not sure it's the same since this doesn't actually happen in Chrome. It also doesn't happen in a vanilla Electron app.

If I try to remove all event listeners from the document before adding the iframe, it still does this craziness. If I log the stack trace, it seems like this causes `iframe.jsx`'s `componentDidMount` to get hit over and over again...

**Comment by flotwig** [comment]
Using the protocol monitor, you can see `{code: -32000, message: "Could not find node with given id"}` errors when hovering over the empty elements, leading me to believe this is some kind of bug with the internal state of the DevTools client:

![image](https://user-images.githubusercontent.com/1151760/148998188-4557d9c1-12c9-49c5-b5ee-c1966bc7a9ee.png)

**Comment by tbiethman** [comment]
Can confirm this is still occurring in develop with Electron bumped to 18.0.4.

**Comment by lmiller1990** [comment]
I'm sad. The electron browser is basically dead on arrival for anything other than running in CI.

I'm guessing if we ever want this fixed we will need to go deep in electron's code base.

**Comment by flotwig** [comment]
I can no longer reproduce this in the latest version of Cypress (Electron 21.0.0). @lmiller1990 are you able to reproduce?

**Comment by lmiller1990** [comment]
Devtools still seems completely "frozen" and not updating for me, I am on Cypress 11.0.1 using component testing. 🤔 

Edit: on macos, I will try linux later and see if it's happening there, too. Also, I did not try e2e yet.

**Comment by patrickcate** [comment]
This is still an issue with Cypress 12.9.0. I've only tested on E2E testing. Running on OSX.

**Comment by lmiller1990** [comment]
This is fixed by #26573!
